lib/matplotlib/rcsetup.py

```diff
@@ -32,20 +33,34 @@
 from cycler import Cycler, cycler as ccycler
 
 
-# The capitalized forms are needed for ipython at present; this may
-# change for later versions.
-interactive_bk = [
-    'GTK3Agg', 'GTK3Cairo', 'GTK4Agg', 'GTK4Cairo',
-    'MacOSX',
-    'nbAgg',
-    'QtAgg', 'QtCairo', 'Qt5Agg', 'Qt5Cairo',
-    'TkAgg', 'TkCairo',
-    'WebAgg',
-    'WX', 'WXAgg', 'WXCairo',
-]
-non_interactive_bk = ['agg', 'cairo',
-                      'pdf', 'pgf', 'ps', 'svg', 'template']
-all_backends = interactive_bk + non_interactive_bk
+# Deprecation of module-level attributes using PEP 562
+_deprecated_interactive_bk = backend_registry.list_builtin(BackendFilter.INTERACTIVE)
+_deprecated_non_interactive_bk = backend_registry.list_builtin(
+    BackendFilter.NON_INTERACTIVE)
+_deprecated_all_backends = backend_registry.list_builtin()
+
+_deprecated_names_and_args = {
+    "interactive_bk": "matplotlib.backends.registry.BackendFilter.INTERACTIVE",
+    "non_interactive_bk": "matplotlib.backends.registry.BackendFilter.NON_INTERACTIVE",
+    "all_backends": "",
+}
+
+
+def __getattr__(name):
+    if name in _deprecated_names_and_args:
+        arg = _deprecated_names_and_args[name]
+        _api.warn_deprecated(
+            "3.9.0",
```

Comment:
We don't put micros here:
```suggestion
            "3.9",
```

Reply:
Done